common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -845,6 +845,14 @@ private static Coordinate[] extractCoordinates(Geometry geometry) {
         return coordinates;
     }
 
+    public static int numPoints(Geometry geometry) throws Exception {
+        String geometryType = geometry.getGeometryType();
+        if (!(Geometry.TYPENAME_LINESTRING.equalsIgnoreCase(geometryType))) {
+            throw new Exception("Unsupported geometry type: " + geometryType);
```

Comment:
Maybe better if throw `IllegalArgumentException`. Please also add in the error message that only LineString type is accepted.

Reply:
Done